source/features/mark-unread.tsx

```diff
@@ -21,22 +41,22 @@ const stateIcons = {
 	}
 };
 
-async function getNotifications() {
+async function getNotifications(): Promise<Notification[]> {
 	const {unreadNotifications} = await browser.storage.local.get({
 		unreadNotifications: []
 	});
 	return unreadNotifications;
 }
 
-function setNotifications(unreadNotifications) {
+async function setNotifications(unreadNotifications: Notification[]): Promise<void> {
```

Comment:
This is being awaited in certain parts of the code so need to add `async` to respect the `@typescript-eslint/promise-function-async` rule.

Reply:
Agree that if it's returning a promise already it can be `await`ed. I was just trying to respect that rule.